### ISSUE TITLE
[MSL] Variables used in interface blocks aren't resolved correctly

### ISSUE THREAD

**Issue by akb825** [KhronosGroup/SPIRV-Cross#179]
When using interface blocks, the generated code references values that don't exist. For example, the input:

```
uniform Transform
{
    mat4 transform;
} block;

in vec3 position;
in vec4 color;

out VertexOut
{
    vec4 color;
} outputs;

void main
{
    gl_Position = block.transform*vec4(position, 1.0);
    outputs.color = color;
}
```

generates the following output:

```
#include <metal_stdlib>
#include <simd/simd.h>

using namespace metal;

struct Transform
{
    float4x4 transform;
};

struct main0_in
{
    float4 color [[attribute(1)]];
    float3 position [[attribute(0)]];
};

struct main0_out
{
    float4 VertexOut_color [[user(locn0)]];
    float4 gl_Position [[position]];
    float gl_PointSize [[point_size]];
    float gl_CullDistance[1];
    float gl_ClipDistance[1] /* [[clip_distance]] built-in not yet supported under Metal. */;
};

vertex main0_out main0(main0_in in [[stage_in]], constant Transform& block [[buffer(0)]])
{
    main0_out out = {};
    out.gl_Position = block.transform * float4(in.position, 1.0);
    outputs.color = in.color;
    out.gl_Position.y = -(out.gl_Position.y);    // Invert Y-axis for Metal
    return out;
}
```

Notice how it's assigning outputs.color instead of VertexOut_color. I got the same results when manually using a struct.

**Comment by HansKristian-ARM** [comment]
Seems like it doesn't handle the distinction between I/O blocks and plain outputs.

**Comment by billhollings** [comment]
Fixed as part of PR #186.
